### DISCUSSION TITLE
Enum not part of MicroPython?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16309] [General]
Enum not part of MicroPython?
```
>>> theNetwork.isconnected()
True
>>> import mip
>>> mip.install("enum")
Installing enum (latest) from https://micropython.org/pi/v2 to /lib
Package not found: https://micropython.org/pi/v2/package/6/enum/latest.json
Package may be partially installed
>>> import enum
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'enum'
>>> 
```

**Comment by scruss** [comment]
nope. There's some discussion in this issue: [Request for package: micropython-enum #269](https://github.com/micropython/micropython-lib/issues/269). A simple implementation is included in the discussion that may work for you.

As ever, check [mim](https://mim.oliverrobson.tech/) if `mip.install("…")` returns **Package not found**

**Comment by Jibun-no-Kage** [comment]
Yeah, I tried 'mip' and when it failed, figured it was out of scope.  I did the typical cheat, created a static class...
```
class MyEnum:
One = 1
Two = 2
```

Cheesy but still just 'named' constants.  Oh for the days of 'C' with the wonderful '#define' statements! LOL

**Comment by bidrohini** [comment]
You can take a look at this thread: https://forum.micropython.org/viewtopic.php?t=9674

**Comment by Jibun-no-Kage** [comment]
Strictly speaking the lack of a true enum in MIcroPython is not a big deal... just a surprise it did not exist, when I tripped over it.
